components/lang/src/main/java/io/cellery/impl/RunTestSuite.java

```diff
@@ -0,0 +1,276 @@
+
+/**
+ * Native function cellery:runTestSuite.
+ */
+@BallerinaFunction(
+        orgName = "celleryio", packageName = "cellery:0.0.0",
+        functionName = "runTestSuite",
+        args = {@Argument(name = "iName", type = TypeKind.RECORD),
+                @Argument(name = "testSuite", type = TypeKind.RECORD)},
+        returnType = {@ReturnType(type = TypeKind.ERROR)},
+        isPublic = true
+)
+public class RunTestSuite extends BlockingNativeCallableUnit {
+
+    private static final String OUTPUT_DIRECTORY = System.getProperty("user.dir") + File.separator + TARGET;
+    private static final Logger log = LoggerFactory.getLogger(CreateCellImage.class);
+
+    private List<CellImage> cellImageList = new ArrayList<>();
+
+    public static void writeToFile(String context, String targetPath) throws IOException {
+
+        File newFile = new File(targetPath);
+        // delete if file exists
+        if (newFile.exists()) {
+            Files.delete(Paths.get(newFile.getPath()));
+        }
+        //create required directories
+        if (newFile.getParentFile().mkdirs()) {
+            Files.write(Paths.get(targetPath), context.getBytes(StandardCharsets.UTF_8));
+            return;
+        }
+        Files.write(Paths.get(targetPath), context.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public void execute(Context ctx) {
+
+        LinkedHashMap nameStruct = ((BMap) ctx.getNullableRefArgument(0)).getMap();
+        final BMap refArgument = (BMap) ctx.getNullableRefArgument(1);
+        BRefType<?>[] tests = ((BValueArray) refArgument.getMap().get("tests")).getValues();
+
+        try {
+            processTests(tests, nameStruct);
+            generateCells();
+        } catch (BallerinaException e) {
+            ctx.setReturnValues(BLangVMErrors.createError(ctx, e.getMessage()));
+        }
+    }
+
+    private void processTests(BRefType<?>[] tests, LinkedHashMap nameStruct) {
+
+        for (BRefType<?> refType : tests) {
+            CellImage cellImage = new CellImage();
+            String name = ((BMap) refType).getMap().get("name").toString();
+
+            cellImage.setCellName(name);
+            cellImage.setOrgName(((BString) nameStruct.get("org")).stringValue());
+            cellImage.setCellVersion(((BString) nameStruct.get("ver")).stringValue());
+
+            if (name.isEmpty()) {
+                break;
+            }
+            Test test = new Test();
+            test.setName(name);
+            LinkedHashMap sourceMap = ((BMap) ((BMap) refType).getMap().get("source")).getMap();
+            test.setSource(sourceMap.get("image").toString());
+            LinkedHashMap envMap = ((BMap) ((BMap) refType).getMap().get("envVars")).getMap();
+            CelleryUtils.processEnvVars(envMap, test);
+            cellImage.setTest(test);
+            cellImageList.add(cellImage);
+        }
+    }
+
+    private void generateCells() {
+
+        for (CellImage cellImage : cellImageList) {
```

Comment:
Duplicate code found in this method and CreateCellImage.generateCells class. Shall we move the duplicate code to utils method?